src/state/apis/snapshot/snapshot.ts

```diff
@@ -152,6 +194,63 @@ export const snapshotApi = createApi({
         }
       },
     }),
+    getThorVotingPower: build.query<FoxVotingPowerCryptoBalance, void>({
+      queryFn: async (_, { dispatch, getState }) => {
+        try {
+          const accountIds: AccountId[] =
+            (getState() as ReduxState).portfolio.accountMetadata.ids ?? []
+          const strategies = await (async () => {
+            const maybeSliceStragies = (getState() as ReduxState).snapshot.thorStrategies
+            if (maybeSliceStragies) return maybeSliceStragies
+
+            const strategiesResult = await dispatch(
+              snapshotApi.endpoints.getThorStrategies.initiate(),
+            )
+            return strategiesResult?.data
+          })()
+          if (!strategies) {
+            console.log('snapshotApi getThorVotingPower could not get strategies')
+            return { data: bn(0).toString() }
+          }
+          const evmAddresses = Array.from(
+            accountIds.reduce<Set<string>>((acc, accountId) => {
+              const { account, chainId } = fromAccountId(accountId)
+              isEvmChainId(chainId) && acc.add(account)
+              return acc
+            }, new Set()),
+          )
+          const delegation = false // don't let people delegate for discounts - ambiguous in spec
+          const votingPowerResults = await Promise.all(
+            evmAddresses.map(async address => {
+              const votingPowerUnvalidated = await getVotingPower(
+                address,
+                '1',
+                strategies,
+                THOR_TIP_014_BLOCK_NUMBER,
+                THORSWAP_SNAPSHOT_SPACE,
+                delegation,
+              )
```

Comment:
Probably never as it's only 2 call then it's even kept in cache somehow as I don't have a new request when I reload the page

We also have a retry, we might want to throttle if we notice any rate limits issues but can we do it in a follow up PR or if we notice an issue?

Reply:
Sweet yeah only 2 addresses should be totally fine. For some reason in my head `evmAddresses` could be like 150 or something but no probs.

Reply:
> Sweet yeah only 2 addresses should be totally fine. For some reason in my head `evmAddresses` could be like 150 or something but no probs.

Ah I think I missed that part, I was thinking about only one call but it effectively do 1 call by address which is already an issue on the fox side

It's obviously an issue as we have 100 request by mins which would fail for 50 EVM addresses, creating an issue for a follow up that I'll do tomorrow if you don't mind (https://github.com/shapeshift/web/issues/8048)